pkg/auth/providers/ldap/ldap_client.go

```diff
@@ -73,6 +73,19 @@ func (p *ldapProvider) loginUser(credential *v32.BasicLogin, config *v3.LdapConf
 		fmt.Sprintf("(%v=%v)", ObjectClass, config.UserObjectClass),
 		operationalAttrList, nil)
 	opResult, err := lConn.Search(searchOpRequest)
+
+	// If no object is found, try again using SA
+	if err != nil && ldapv3.IsErrorWithCode(err, ldapv3.LDAPResultNoSuchObject) {
```

Comment:
I check for this error because it's the error returned by my LDAP provider in this situation. @digiserg also faced this error code when using `ldapsearch` as he confirmed in PR #34951. Should we re-bind the SA in case of `err != nil` without checking for this specific error?

Reply:
This is because some LDAP servers with heightened security (in my case, JumpCloud) don't let a user perform searches even to look up its own entry. This is not considered an error and is why after validating a user's password the server should bind back into the service account and perform any subsequent queries using the service account as the user most likely again lacks permissions to do any day to day operations.

The right approach is to:

- verify bind credentials
- look up user DN
- verify user credentials (bind as the user)
- re-bind with the service account
- perform any subsequent searches

Reply:
Talked about the approach offline. I think this error code has a widely-enough used meaning that we can check for it here. I'll resolve this thread - you don't need to make any changes to this. Thank you for explaining.